pkg/process/net/uds_test.go

```diff
@@ -72,3 +77,106 @@ func TestNewUDSListener(t *testing.T) {
 		testWorkingNewUDSListener(tt, dir+"/net.sock")
 	})
 }
+
+type fakeHandler struct {
+	t       *testing.T
+	request string
+}
+
+func (f *fakeHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	f.request = req.URL.String()
+	w.WriteHeader(200)
+}
+
+func testHttpServe(t *testing.T, shouldFailed bool, f *fakeHandler, prefixCmd []string, uid int, gid int) (err error) {
+	dir := t.TempDir()
+	err = os.Chmod(dir, 0777)
+	require.NoError(t, err)
+	err = os.Chmod(dir+"/..", 0777)
+	require.NoError(t, err)
+
+	socketPath := dir + "/test.http.sock"
+	conn, err := net.Listen("unix", socketPath)
+	require.NoError(t, err)
+	t.Cleanup(func() { _ = conn.Close() })
+	err = os.Chmod(socketPath, 0666)
+	require.NoError(t, err)
+
+	go func() {
+		cmd := append(prefixCmd, "curl", "-s", "--unix-socket", socketPath, "http://unix/test")
+		o, err := exec.Command(cmd[0], cmd[1:]...).CombinedOutput()
+		conn.Close() // closing the server
+		if !shouldFailed {
+			if err != nil {
+				t.Log(cmd, string(o))
+			}
+			require.NoError(t, err)
+		} else {
+			require.Error(t, err)
+		}
+	}()
+
+	return HttpServe(conn, f, uid, gid)
+}
+
+func lookupUser(t *testing.T, name string) (usrID int, grpID int, usrIDstr string, grpIDstr string) {
+	usr, err := user.Lookup(name)
+	require.NoError(t, err)
+
+	usrID, err = strconv.Atoi(usr.Uid)
+	require.NoError(t, err)
+
+	grpID, err = strconv.Atoi(usr.Gid)
+	require.NoError(t, err)
+
+	grp, err := user.LookupGroupId(usr.Gid)
+	require.NoError(t, err)
+
+	return usrID, grpID, usr.Username, grp.Name
+}
+
+func checkIfSudoExistAndNotInteractive(t *testing.T) {
+	sudo := exec.Command("/tmp/s.sh", "id")
```

Comment:
Where does `/tmp/s.sh` come from ? Shouldn't it be `sudo` ?

Reply:
sorry was a local test